AK/IpAddressCidr.h

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2024, famfo <famfo@famfo.xyz>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/IPv4Address.h>
+#include <AK/IPv6Address.h>
+#include <AK/Result.h>
+
+namespace AK {
+
+class [[gnu::packed]] IPv4AddressCidr;
+class [[gnu::packed]] IPv6AddressCidr;
+template<typename Address>
+class [[gnu::packed]] AddressTraits;
+
+template<OneOf<IPv4AddressCidr, IPv6AddressCidr> AddressFamily>
+class [[gnu::packed]] IPAddressCidr {
```

Comment:
In all cases, using `[[gnu::packed]]` is unnecessary here, since this type is not transmitted over the network and we don’t need particular small alignment guarantees. This is only done for the other IP-related types that actually go out to hardware.